### PR TITLE
NewSidebar performance improvements

### PR DESCRIPTION
**Pull request by luluiz**
**User-Facing Changes**

N/A

**Description**
- Moved type to the proper file
- Moved style to the proper file
- Used useCallback hook to improve performance reducing unnecessary renderings.
- Used useMemo hook to improve performance reducing unnecessary processing.
- Unified NewSidebarItem and SidebarItem.

**Checklist**

- [x] The web version was tested and it is running ok
- [x] The desktop version was tested and it is running ok
- [ ] This change is covered by unit tests
- [x] Files constants.ts, types.ts and *.style.ts have been checked and relevant code snippets have been relocated

### CODE REVIEW

File: packages/suite-base/src/components/Sidebars/index.tsx
Comment (aneuwald-ctw) [inline packages/suite-base/src/components/Sidebars/index.tsx:1]: @luluiz as you are making changes on the sidebar, I would (if it is possible and not very time-consuming) remove the old sidebar and leave only the "NewSidebar", to not have this duplication on our codebase.

**Review by aneuwald-ctw — COMMENTED** [review]
Just a suggestion, the rest seams ok for me

File: packages/suite-base/src/components/Sidebars/index.tsx
Comment (luluiz) [reply packages/suite-base/src/components/Sidebars/index.tsx:1]: nice, I'm doing it

Comment (luluiz) [reply packages/suite-base/src/components/Sidebars/index.tsx:1]: done

**Pull request merged**